src/live-programming.rst

```diff
@@ -26,6 +26,64 @@ Interactive Debuggers
+
+.. container:: bib-item
+
+  .. bibliography:: live-programming.bib
+    :filter: key == "sorva2013review"
+
+ This paper serves as a survey of program visualization systems whose **task** is to aid the learning and teaching of introductory programming, with an intended **audience** of novice programmers and *CS1* teachers. Even a visualization that has been painstakingly crafted to be as lucid as possible may fail to aid learning in practice. Therefore, the paper introduces a new taxonomy to describe how program visualization systems engage learners.
+
+.. container:: bib-item
+
+  .. bibliography:: live-programming.bib
+    :filter: key == "murphy2010interactive"
+
+ **Code smells** are characteristics of software that indicate that code may have a design problem. The paper propose a novel smell detector, Stench Blossom, that provides an iteractive ambient visualization designed to first give programmers a quick, high-level overview of the smells, and then to help in understanding the smells if users wish. As a result, the experiment confirmed that programmers identify more smells and make more confident and informed refactoring jedgements using the tool than not using the tool.
+
+.. container:: bib-item
+
+  .. bibliography:: live-programming.bib
+    :filter: key == "reiss2005paradox"
+
+ This paper argues that most past and current work in the field is out of touch with the reality of software development and that new approaches and new ideas are needed.
+
+.. container:: bib-item
+
+  .. bibliography:: live-programming.bib
+    :filter: key == "de2002visualizing"
+
+ Jinsight is a tool for exploring a Java program's runtime behavior visually, featuring a collection of linked views.
+
+  - a basic visualization of resource consumption in terms of classes, instances, and methods
+  - a reference pattern view with extraneous detail eliminated that could detect memory leak
+  - a performance analysis by visualizing event sequences
+
+ With Jinsight, users have successfully diagnosed numerous problems on large commercial applications. However, the visualization is specialized for particular tasks and the visualized application could only run on a single *JVM*.
+
+.. container:: bib-item
+
+ .. bibliography:: live-programming.bib
+   :filter: key == "goodall2010visual"
+
+ This paper describes a system that brings together the results of disparate software analysis tools into a visual environment to support the triage and exploration of code vulnerabilities. The system could give more confidence that the detected vulnerabilities are not false positive by correlates and normalizes the output of multiple software analysis tools. And the user may also wish to associate the vulnerability with the programmer who regularly checks in the code with vulnerabilities or the main developer that modifiers more code than anyone else. This workflow allows the system to scale to large code bases with tens of thousands of vulnerabilities.
```

Comment:
* This may not exactly fall under "program visualization", since what is being visualized is the result of a program analysis. However, you can leave it here for now.
* "by correlates and normalizes" -> "by correlating and normalizing"
* "And the user may also wish to " -> "The user can also "
* Not sure I understand "the programmer who regularly checks in the code with vulnerabilities or the main developer that modifiers more code than anyone else"? is it trying to assign "blame" for each vulnerability in some heuristic way? think about a way to describe this more clearly